peg_src/jsdoctype.pegjs

```diff
@@ -59,10 +59,10 @@ JsIdentifier = $([a-zA-Z_$][a-zA-Z0-9_$]*)
 
 // It is transformed to remove left recursion.
 // See https://en.wikipedia.org/wiki/Left_recursion#Removing_left_recursion
-NamepathExpr = rootOwner:(ParenthesizedExpr / TypeNameExpr) memberPartWithOperators:(_ InfixNamepathOperator _ MemberName)* {
+NamepathExpr = rootOwner:(ParenthesizedExpr / TypeNameExpr) memberPartWithOperators:(_ InfixNamepathOperator _ "event:"? _ MemberName)* {
```

Comment:
Thanks for the contribution!

I think `event:` should be similar to `external:` that defined as [`ExternalNameExpr`](https://github.com/Kuniwak/jsdoctypeparser/blob/8f0bf03b1e658d8a9956f79e069cfdd64c027428/peg_src/jsdoctype.pegjs#L185-L195).  Could you define a new expression that is like `EventNameExpr` instead of extending the `NamepathExpr`?

Reply:
Sure, and thanks for the timely response... Will try to take a look to change now...

Reply:
I guess I'm at a loss on how to do this... (And it also would need to be done for `ModulePathExpr`)...

If it is an event name expression, it would seem to imply there should also be non-event name expression. The way I treated it was not to give it its own `NodeType`, though I guess it should probably have its own type. In any case, if it is a new expression, I'm not sure how to swap it in at that point *within* the expression...

It's a bit different from `external:` as this "event:" is not preceding the whole `NamepathExp`, but is just a portion within it... It seems I'd have to reimplement `NamepathExpr` but with the prefix to avoid putting it in `NamepathExpr`. Is that what you'd like?

Reply:
@brettz9 

> It's a bit different from external: as this "event:" is not preceding the whole NamepathExp, but is just a portion within it...

You're right. I had misunderstood the syntax of `event:`...
And now, I feel your implementation approach is almost right. And I feel it is better that we can know whether the member has a event prefix or not. So a new flag indicating it should be placed on `ModulePathExpr`. How do you feel it?

Reply:
Very good to include it I think—especially since discovering that per http://usejsdoc.org/tags-event.html:

> JSDoc automatically prepends the namespace event: to each event's name. In general, you must include this namespace when you link to the event in another doclet. (The @fires tag is a notable exception; it allows you to omit the namespace.)

Reply:
Plan to prepare now...

Reply:
I think `hasEventPrefix` is bitty better. because the `isEvent` for some event components can be  `false` in another syntax such as JSDuck/ClosureLibrary (because it do not have to have `event:` prefix on another syntax).

Reply:
I've published an update using `hasEventPrefix`, and also added it to the README and to the default publisher...

Reply:
Actually just now, I amended again, this time to add a couple publishing tests, one for members (as there had been none) and one for an event member... I plan to stop now though...

Reply:
Is there anything you need further on my end?